host/windows/syscall.c

```diff
@@ -692,6 +1420,18 @@ static oe_host_fd_t _dup_socket(oe_host_fd_t oldfd)
     return -1;
 }
 
+static bool _is_socket(oe_host_fd_t oldfd)
+{
+    if (!oldfd || oldfd >= 0 && oldfd <= 2) 
+    {
+        return false;
+    }
```

Comment:
_get_socket will check the magic number of on the file descriptor structure, do I don't think you need this if check.